Added modal for calendar events. Adjusted CSS for calendar and event modal to be more flexible with different browser sizes.

nnguyen144 opened the pull request:
Modals are web-page elements that display in front of other web-page elements and disables them as well. I used modals to display the details of events that are located on the calendar.

![piroC0g](https://github.com/user-attachments/assets/3a3289f2-bcc5-4c25-8ea4-2522fdc4fc09)

As seen in the gif above, events displayed on the calendar can be clicked on, and when clicked on, a modal pops up that displays details of the event. Only the title and start time of the event are required, while the location, end time, and description are optional. If the title or description are too long to be displayed as a whole, there are scrollbars that can be used to navigate through the entirety of each text. Clicking outside the modal or clicking on the close button marked by the "X" in the top right will close the current event modal.

Another feature that was added was CSS to improve how the calendar appears when the browser size changes. Below is an example where the browser is horizontally shortened and expanded to show how the calendar appears.
![2xydZfL](https://github.com/user-attachments/assets/d14a330f-8d29-4e48-ae82-077c709a5ad3)

Below is how the browser appears when vertically changed.
![UHahHRh](https://github.com/user-attachments/assets/1868c860-acc9-43a6-b78e-43f8e32f89ad)


And below is how the calendar appears when vertically and horizontally changed.
![opera_H17B5Il21e](https://github.com/user-attachments/assets/28d288ed-111d-4e6f-82c5-5fbb19f30f09)

CSS testing was done on a 13" laptop and a 1920 x 1080 monitor. 

The reason why there is space on the left side of the calendar is for a potential sidebar component that can be added in the future. I will also add space on the top of the calendar if needed for a header component.

If there are any changes that should be made to what I have so far (for example, are there any event details that I could include?), let me know.

- Nathan Nguyen

nnguyen144 committed:
Fixed Calendar.js to show EventModal on click. Added EventModal.js and css. Will expand on them after

nnguyen144 committed:
Adjusted the CSS of the Calendar and Event Modal to make them more responsive to different browser sizes.

nnguyen144 committed:
Fixed formatting of hours and minutes for events

nnguyen144 committed:
Merge branch 'UNLV-CS472-672:master' into calendar-modal

nnguyen144 committed:
Added more CSS changes to improve frontend

misherwa commented on frontend/src/components/Calendar/Calendar.module.css:
```diff
@@ -79,17 +86,17 @@
 .day {
     padding: 2px; 
-    border: 1px solid #880808; 
+    border: 1px solid #CF0A2C; 
```
If this is the class to edit the calendar box days in the Calendar.module.css file, perhaps change the background color inside of the boxes. Maybe to white or grey?

you could do this by adding this line "background-color: white;" to change it to white

or this line "background-color: #f0f0f0;" to change it to grey.

Personally, it looks weird having the background of the boxes be the same as the background color of the website. So changing it to white or grey could help to make the calendar pop out.

nnguyen144 replied:
Hi Aarush, you make a good point about the background color of the website as of right now not blending in well with the calendar boxes. Although I believe that it would be fine as it is now if the background color of the website globally was switched to white or any white-variation kind of color instead of black.

Example:
![image](https://github.com/user-attachments/assets/90d690a5-826e-4e40-a6fd-793760d7e226)

I switched the website to have a white background (#f2f2f2) and kept the boxes to have the same background color as the website. In my opinion, it looks better like this. Keep in mind that the LongTitle... event is a bright green and so it looks awkward on a white background, however I do not intend to have that bright of a color to be an option for when a user chooses their event color when I set up my CRUD operations for the events. Let me know of what you think of the design in the screenshot above.

- Nathan Nguyen

nnguyen144 replied:
Either the background color of the website and the background color of the boxes are the same, or maybe the boxes are slightly different in color than the website. Either option I think is the best.

misherwa replied:
> Hi Aarush, you make a good point about the background color of the website as of right now not blending in well with the calendar boxes. Although I believe that it would be fine as it is now if the background color of the website globally was switched to white or any white-variation kind of color instead of black.
> 
> Example: ![image](https://private-user-images.githubusercontent.com/50503296/380040917-90d690a5-826e-4e40-a6fd-793760d7e226.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4ODYzNjMsIm5iZiI6MTcyOTg4NjA2MywicGF0aCI6Ii81MDUwMzI5Ni8zODAwNDA5MTctOTBkNjkwYTUtODI2ZS00ZTQwLWE2ZmQtNzkzNzYwZDdlMjI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDE5NTQyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhMWZhMjA1YzFkYTZhNWMyMjZkMzg0NTQ3OWIwODk1ZjYxNjJiN2Y5NjI0NDZhM2I4YjRiZDI2NDY4M2EzOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YoMR6sH69hhqPdhwM8C6x5Q3ENEj2lILOKifxtKAhg8)
> 
> I switched the website to have a white background (#f2f2f2) and kept the boxes to have the same background color as the website. In my opinion, it looks better like this. Keep in mind that the LongTitle... event is a bright green and so it looks awkward on a white background, however I do not intend to have that bright of a color to be an option for when a user chooses their event color when I set up my CRUD operations for the events. Let me know of what you think of the design in the screenshot above.
> 
> * Nathan Nguyen

Hey Nathan, I started messing around with the calendar's coding to understand it better. I was able to make a page that looks like this:

![s2](https://github.com/user-attachments/assets/0b628f4c-a627-484f-b0d2-8e1ae0a78c26)

Again, this is just a nitpick, but if you want, I can tell you what you need to add or change. It's pretty simple, so I'll just add comments on the "Files Changed" page.

Also, I noticed in your globals.css file, you have that ":root" on line 3. As you can see on line 2 from this image below, I think you can delete lines 3 to 6. 
![q2](https://github.com/user-attachments/assets/dd151da0-5a37-439b-a0be-5d8c181d8cff)

I tried changing the background and foreground colors with that first ":root" but it seems to not be working. But if you see on line 10, I changed the color to #a9a9a9 (Medium Grey) for the background of  the website

misherwa commented:
Hey Nathan, the vertical and horizontal size changes look nice, and thanks for a visual example of it too. 

Sorry, I'm having trouble opening/getting to the calendar page. I'm using Codespace and this is what my screen looks like:
![q](https://github.com/user-attachments/assets/caa50357-a7ee-4acf-adb0-55cca20924f8)

Could you please tell me the subdirectory or what the URL should look like? Currently, mine looks like "https://silver-trout-v6gg56q7wxr6hxgr5-3000.app.github.dev/"


A quick nitpick from me, I feel like you could change the background color inside of the boxes. Maybe to white or grey? I mentioned it in the comment above. But I'll try to test out the colors white and black once I get the calendar page running on Codespace.

nnguyen144 commented:
Hey Aarush,

>Sorry, I'm having trouble opening/getting to the calendar page.

I uploaded the calendar as a component as we don't have all the necessary components to make a full calendar page yet. If you want to test out the calendar yourself, then you can make a page file (or edit the page.js file which you currently have open on your screen so you don't need to access other links), import the Calendar.js file, and put the Calendar into the HTML.

I'm unsure of how everyone else does it, but I have the page.js file locally set up like this so I can work on my calendar
![image](https://github.com/user-attachments/assets/b5f75b0f-15c6-4e88-bfa9-33f1d9f975b3)

barrioskevin commented:
Hi, I like that fact that you are making iterations to your calendar component. It's looking really good I just have a few suggestions if I may. 

The calendar is offset so it sticks to the right hand side of the page. I don't know if this was intended but it looks a little unpleasing. You can fix this easily by adding in `margin-right: auto,` right above or below where you have `margin-left: auto` in your `Calendar.module.css` file under `.calendar`.

The view for when the page shrinks too far also looks a little odd but I suppose if it's needed to show the entire calendar it's not too bad. 

The modal that you added is really cool! However I think you installed a new dependency [https://www.npmjs.com/package/react-modal](url) so make sure that you include any changes to `package.json` and `package-lock.json` in order for us to be able to build and run the project with everything needed included. 

I also think that you should be opening up specific issues so that the rest of the group has a better idea of your goals and what you are working on. 

Right now the events in the backend is lacking a lot of features and functionality. Events don't even have a user assigned to them so right now all of the events you are working with are 'global'. I wouldn't try making to much progress on the front end until we have a finalized idea of how the events feature will work. 

I think as long as you include the updated package files this PR should be okay to merge. It includes extra progress made to your work and it's an updated version that we can fall back on. It's just that I think we should be opening issues before working on things like this. Thank you for your time.

nnguyen144 committed:
Added react-modal dependency to packages

nnguyen144 committed:
Adjusted calendar width

courtmcgee2197 commented:
This looks like a great start to the calendar feature. Visually I prefer a brighter background for a calendar as I want to be able to scan it quickly. Also maybe calendar should be centered for visual appeal.

Here are some things to consider with the calendar: 
Since the calendar is suppose to be user specific how will we go about updating the users calendar with global events while maintaining the users own events. Typically calendars will also have a list view of every event or important date on the calendar below it allowing them to read the details of each event possibly with timestamps. Overall great start for front end visuals.

nnguyen144 commented:
Hello Kevin,

>The calendar is offset so it sticks to the right hand side of the page. I don't know if this was intended but it looks a little unpleasing. You can fix this easily by adding in margin-right: auto, right above or below where you have margin-left: auto in your Calendar.module.css file under .calendar.

I intended that so I could leave space on the left side for a sidebar if we ever choose to implement one, but because we don't have one right now, I've gone ahead and fixed it to expand the entire width of the browser as per your suggestion.

>The modal that you added is really cool! However I think you installed a new dependency [https://www.npmjs.com/package/react-modal](https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub/pull/url) so make sure that you include any changes to package.json and package-lock.json in order for us to be able to build and run the project with everything needed included.

That is my mistake. I've included them in my most recent commits now. Thank you.

>I also think that you should be opening up specific issues so that the rest of the group has a better idea of your goals and what you are working on.

I will make sure I open issues whenever I implement PRs from now on. Thank you.

>Right now the events in the backend is lacking a lot of features and functionality. Events don't even have a user assigned to them so right now all of the events you are working with are 'global'. I wouldn't try making to much progress on the front end until we have a finalized idea of how the events feature will work.

I'll be working on these kings of things for the backend in my next PR, which includes differentiating between a user event and hub events, where user events are personal and can only be seen by the user that created it, and hub events are seen by those who are subscribed to certain hubs.

>I think as long as you include the updated package files this PR should be okay to merge. It includes extra progress made to your work and it's an updated version that we can fall back on. It's just that I think we should be opening issues before working on things like this. Thank you for your time.

I've fixed the calendar width and included the dependencies, so if you feel like my code is adequate enough to merge, then feel free so. Thank you for the review.

-Nathan Nguyen

misherwa commented:
> Hey Aarush,
> 
> > Sorry, I'm having trouble opening/getting to the calendar page.
> 
> I uploaded the calendar as a component as we don't have all the necessary components to make a full calendar page yet. If you want to test out the calendar yourself, then you can make a page file (or edit the page.js file which you currently have open on your screen so you don't need to access other links), import the Calendar.js file, and put the Calendar into the HTML.
> 
> I'm unsure of how everyone else does it, but I have the page.js file locally set up like this so I can work on my calendar ![image](https://private-user-images.githubusercontent.com/50503296/380036892-b5f75b0f-15c6-4e88-bfa9-33f1d9f975b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4ODYzNjMsIm5iZiI6MTcyOTg4NjA2MywicGF0aCI6Ii81MDUwMzI5Ni8zODAwMzY4OTItYjVmNzViMGYtMTVjNi00ZTg4LWJmYTktMzNmMWQ5Zjk3NWIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI1VDE5NTQyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0OGEwODlhMDAxZjQ3MTA2YzIxOWNlYWM4M2Q2MWRjNzE3NDQzN2Q3MjNmMGE0YzQzMjlhZWMxMmM2NjUyNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.R9F1I7G-cClQ2PBQ2utOgCzqm1C8eJCnalk864PDvgg)

Gotcha, thanks a lot, I was able to open up the calendar page.

misherwa commented:
Hey Nathan, I started messing around with the calendar's coding to understand it better. I was able to make a page that looks like this:

![s2](https://github.com/user-attachments/assets/0b628f4c-a627-484f-b0d2-8e1ae0a78c26)

Again, this is just a nitpick, but if you want, I can tell you what you need to add or change. It's pretty simple, so I'll just add comments on the "Files Changed" page if you want.

Also, I noticed in your globals.css file, you have that ":root" on line 3. As you can see on line 2 from this image below, I think you can delete lines 3 to 6. 
![q2](https://github.com/user-attachments/assets/dd151da0-5a37-439b-a0be-5d8c181d8cff)

I tried changing the background and foreground colors with that first ":root" but it seems to not be working. But if you see on line 10, I changed the color to #a9a9a9 (Medium Grey) for the background of  the website

nnguyen144 commented:
Thank you for the advice Aarush. Feel free to merge my PR after you read the other comments and see if the code accomplishes its purpose.

nnguyen144 committed:
Merge branch 'master' into calendar-modal
